src/blocks/Block.as

```diff
@@ -1017,4 +1018,49 @@ public class Block extends Sprite {
 		return s.replace(/^/gm, "    ");
 	}
 
+	/* Blocking */
+
+	public function resetBlockingCount():void {
+		// should be called for a top stack block - resets all blocks in stack
+		resetArgsBlockingCount();
+		if(subStack1) subStack1.resetBlockingCount();
+		if(subStack2) subStack2.resetBlockingCount();
+		if(nextBlock) nextBlock.resetBlockingCount();
+	}
+
+	public function resetArgsBlockingCount():void {
+		// descend through all block's args, resetting to zero
+		blockingCount = 0;
+		var args:Array = this.args;
```

Comment:
I'd prefer not to make a local variable that hides a class member like this. In this particular case, I think this function is called seldom enough that copying the property to a local variable is not likely to have a high performance impact, so I would just remove the local variable. If you've checked into the performance and it turns out that I'm wrong -- certainly possible :) -- I would recommend renaming the variable to avoid shadowing the member.